src/couch_dist/src/couch_dist.erl

```diff
@@ -0,0 +1,106 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_dist).
+
+-export([childspecs/0, listen/2, accept/1, accept_connection/5,
+    setup/5, close/1, select/1, is_node_name/1]).
+
+
+childspecs() ->
+    {ok, [{ssl_dist_sup, {ssl_dist_sup, start_link, []},
+        permanent, infinity, supervisor, [ssl_dist_sup]}]}.
+
+
+listen(NodeName, Host) ->
+    Name = case is_atom(NodeName) of
+               true -> atom_to_list(NodeName);
+               false -> NodeName
+           end,
+    case no_tls(Name ++ "@" ++ Host) of
+        true -> inet_tcp_dist:listen(Name, Host);
+        false -> inet_tls_dist:listen(Name, Host)
+    end.
+
+
+accept(Listen) ->
+    case no_tls(node()) of
+        true -> inet_tcp_dist:accept(Listen);
+        false -> inet_tls_dist:accept(Listen)
+    end.
+
+
+accept_connection(AcceptPid, DistCtrl, MyNode, Allowed, SetupTime) ->
+    case no_tls(MyNode) of
+        true -> inet_tcp_dist:accept_connection(
+            AcceptPid, DistCtrl, MyNode, Allowed, SetupTime);
+        false -> inet_tls_dist:accept_connection(
+            AcceptPid, DistCtrl, MyNode, Allowed, SetupTime)
+    end.
+
+
+setup(Node, Type, MyNode, LongOrShortNames, SetupTime) ->
+    case no_tls(Node) of
+      true -> inet_tcp_dist:setup(
+          Node, Type, MyNode, LongOrShortNames, SetupTime);
+      false -> inet_tls_dist:setup(
+          Node, Type, MyNode, LongOrShortNames, SetupTime)
+    end.
+
+
+close(Socket) ->
+    inet_tls_dist:close(Socket).
+
+
+select(Node) ->
+    inet_tls_dist:select(Node).
+
+
+is_node_name(Node) ->
+    inet_tls_dist:is_node_name(Node).
+
+
+no_tls(NodeName) when is_atom(NodeName) ->
+    no_tls(atom_to_list(NodeName));
+
+no_tls(NodeName) when is_list(NodeName) ->
+    try
+        {ok, Args} = init:get_argument(couch_dist),
+        GlobPatterns = [V || [K, V] <- Args, K == "no_tls"],
+        lists:any(fun(P) -> match(NodeName, P) end, GlobPatterns)
+    catch
+        error:_ -> false
```

Comment:
Is the `error:_` to catch the case when there are `couch_dist` no arguments specified, or some failure in re match expression compilation?

If we mean to handle the case when `couch_dist` args are not specified we should explicitly check for that case as opposed to relying to catch an `error:badmatch` failure from `{ok, Args} =`.

It would be something like:

```erlang
case init:get_argument(couch_dist) of
    {ok, Args} ->
        % ... do the matching
         ...;
     error ->
          false
end.
```

Reply:
Changed, thanks for the review @nickva